backend/controllers/postController.js

```diff
@@ -0,0 +1,110 @@
+const {Post} = require('../model/db')
+const {tokenDecoder} = require('../util/tokenDecoder')
+const multer = require('multer')
+const path = require('path');
+//modifies storage path and file name
+const storage = multer.diskStorage({
+	destination: './uploads/',
+	filename: function(req, file, cb){
+		cb(null, file.originalname);
+  }
+});
+
+var upload = multer({storage : storage});
+
+//Adds new posts to database also adds uploaded files to filesystem
+addPost = async function(req,res) {
+	decodedToken = tokenDecoder(req)
```

Comment:
instead of doing it in every controller function, we can use middlewares that do it for all of the controller functions, please revise.

Reply:
yeah, middleware is better, i updated it